tests/test_cases/test_cocotb/test_timing_triggers.py

```diff
@@ -256,5 +256,5 @@ async def test_neg_timer(dut):
         Timer(-42)  # no need to even `await`, constructing it is an error
     # handle 0 special case
     with warnings.catch_warnings(record=True) as w:
-        Timer(0)
+        await Timer(0)
```

Comment:
There is no need to `await` here, the warning is emitted without it.

Reply:
(the test is failing because you are using `self.log.warn` and not `warnings.warn`)

Reply:
yes but I felt it would make more sense as construction does succeed in this case and this is how it would be used practically, but I don't care. (I'm aware this is not the failure cause)

Reply:
will revert if you wish

Reply:
When doing warning tests, it's best to put as little code as possible in the context manager. In principle we could do
```
with warnings.catch_warnings(record=True) as w:
    t = Timer(0)
    assert ...

await t
```
which would both check that the warning is emitted, and that the resulting trigger works. Remember though that the whole point in the warning is that on some simulators `await Timer(0)` does _not_ work, and so if we tested that our tests would fail.